MericCelik/assignment3/Dictionary.java

```diff
@@ -0,0 +1,80 @@
+import java.util.HashSet;
+
+/**@author AycaMericCelik
+ * */
+
+public class Dictionary {
+    private String[] words;
+    private String[] prefixes;
+
+    public Dictionary(String[] words, String[] prefixes){
+        this.words = words;
+        this.prefixes = prefixes;
+    }
+
+    /**Finds all prefixes of given array's words. It gets all the substrings of a string.
+     * Complexities: Best O(1) when it has one element with one character.
+     *              Worst O(N*M*L), Average O(N*M*L), N is word count, M is average complexity of
+     *              constructing a substring and L is the average number of word's substring.
+     * @param words: array to find its element's substrings.
+     * @return set of prefixes(no duplicate)
+     * @throws NullPointerException when the argument is invalid.
+     * */
+    public String[] getPrefixes(String[] words){
+        if(words == null)
+            throw new NullPointerException("Words field is null!");
+        HashSet<String> prefixset = new HashSet<>();
+        int wordsize;
+        for(String word:words){
+            wordsize = word.length();
+            for(int i=1; i<wordsize+1; i++){
+                prefixset.add(word.substring(0,i));
+            }
+        }
+        return prefixset.toArray(new String[prefixset.size()]);
+    }
+
+    public Dictionary(String[] words){
```

Comment:
I'd say it would be acceptable to have this constructor as the only public constructor. :) You can remove the other one.